src/Rotations.jl

```diff
@@ -0,0 +1,199 @@
+@doc doc"""
+    Rotations{N} <: Manifold
+
+Special orthogonal manifold $\mathrm{SO}(n)$ represented by $n\times n$
+real-valued orthogonal matrices with determinant $+1$.
+
+# Constructor
+
+    Rotations(n)
+
+generates the $\mathrm{SO}(n) \subset \mathbb R^{n\times n}$
+"""
+struct Rotations{N} <: Manifold end
+Rotations(n::Int) = Rotations{n}()
+
+@doc doc"""
+    manifold_dimension(S::Rotations)
+
+returns the dimension of the manifold $\mathrm{SO}(n)$, i.e. $\frac{n(n-1)}{2}$.
+"""
+manifold_dimension(S::Rotations{N}) where {N} = div(N*(N-1), 2)
```

Comment:
I'd make this a generated function, since its value is known at compile time, and we may check dimension often.

Reply:
In such simple cases the compiler can automatically detect that the value is constant, so the `@generated` annotation doesn't really help.

Reply:
Oh, good to know!